mediaflow_proxy/utils/http_utils.py

```
@@ -272,15 +272,379 @@ async def request_with_retry(method: str, url: str, headers: dict, **kwargs) ->
+                await send({"type": "http.response.body", "body": b"", "more_body": False})
+            except (httpx.RemoteProtocolError, h11._util.LocalProtocolError) as e:
+                # Handle connection closed errors
+                if data_sent:
+                    # We've sent some data to the client, so try to complete the response
+                    logger.warning(f"Remote protocol error after partial streaming: {e}")
+                    try:
+                        await send({"type": "http.response.body", "body": b"", "more_body": False})
+                        logger.info(
+                            f"Response finalized after partial content ({self.actual_content_length} bytes transferred)"
+                        )
+                    except Exception as close_err:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)